File: CS/DXSample/ViewModels/AppointmentViewModel.cs
using System;
using DevExpress.Mvvm;

namespace DXSample.ViewModels {
    public class AppointmentViewModel : ViewModelBase {
        public int Id {
            get { return GetValue<int>(); }
            set { SetValue(value); }
        }
        public DateTime StartTime {
            get { return GetValue<DateTime>(); }
            set { SetValue(value); }
        }
        public DateTime EndTime {
            get { return GetValue<DateTime>(); }
            set { SetValue(value); }
        }
        public string Subject {
            get { return GetValue<string>(); }
            set { SetValue(value); }
        }
        public int ResourceId {
            get { return GetValue<int>(); }
            set { SetValue(value); }
        }
    }
}

File: VB/DXSample/ViewModels/AppointmentViewModel.vb
Imports DevExpress.Mvvm

Namespace DXSample.ViewModels

    Public Class AppointmentViewModel
        Inherits ViewModelBase

        Public Property Id As Integer
            Get
                Return GetValue(Of Integer)()
            End Get

            Set(ByVal value As Integer)
                SetValue(value)
            End Set
        End Property

        Public Property StartTime As Date
            Get
                Return GetValue(Of Date)()
            End Get

            Set(ByVal value As Date)
                SetValue(value)
            End Set
        End Property

        Public Property EndTime As Date
            Get
                Return GetValue(Of Date)()
            End Get

            Set(ByVal value As Date)
                SetValue(value)
            End Set
        End Property

        Public Property Subject As String
            Get
                Return GetValue(Of String)()
            End Get

            Set(ByVal value As String)
                SetValue(value)
            End Set
        End Property

        Public Property ResourceId As Integer
            Get
                Return GetValue(Of Integer)()
            End Get

            Set(ByVal value As Integer)
                SetValue(value)
            End Set
        End Property
    End Class
End Namespace
